### PR TITLE
test: fix delv trust anchors location on Ubuntu

### PR DESCRIPTION
**Pull request by mrc0mmand**
delv on Ubuntu defaults to /etc/bind/bind.keys instead of /etc/bind.keys
when reading trust anchors, so let's create a symlink to make the test
work there as well.

Resolves: #24453

---

With this PR the test finally passes on Ubuntu Jammy (at least on my local VM). /cc @surajkrishnan14

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by surajkrishnan14** [comment]
Thanks @mrc0mmand  - I tried this out, and ran into a different error. Attached logs:
[logs4.zip](https://github.com/systemd/systemd/files/9435170/logs4.zip)

**Comment by mrc0mmand** [comment]
> Thanks @mrc0mmand - I tried this out, and ran into a different error. Attached logs: [logs4.zip](https://github.com/systemd/systemd/files/9435170/logs4.zip)

Ah, thank you. I managed to break the test on its first run, re-runs were working fine. I pushed an update which should, hopefully, fix this.

**Comment by surajkrishnan14** [comment]
> > Thanks @mrc0mmand - I tried this out, and ran into a different error. Attached logs: [logs4.zip](https://github.com/systemd/systemd/files/9435170/logs4.zip)
> 
> Ah, thank you. I managed to break the test on its first run, re-runs were working fine. I pushed an update which should, hopefully, fix this.

Hmm, I think I'm still seeing the same error (can share logs if needed, LMK). I tried a clean build on commit eb1f34925e9c2952994f6bb5f0bcbcfe74838399

**Comment by mrc0mmand** [comment]
> > > Thanks @mrc0mmand - I tried this out, and ran into a different error. Attached logs: [logs4.zip](https://github.com/systemd/systemd/files/9435170/logs4.zip)
> > 
> > 
> > Ah, thank you. I managed to break the test on its first run, re-runs were working fine. I pushed an update which should, hopefully, fix this.
> 
> Hmm, I think I'm still seeing the same error (can share logs if needed, LMK). I tried a clean build on commit [eb1f349](https://github.com/systemd/systemd/commit/eb1f34925e9c2952994f6bb5f0bcbcfe74838399)

Hmpf, you're right, interesting that it worked the ~5 times I tried it before pushing... let me dig a bit deeper.

**Comment by mrc0mmand** [comment]
Interesting. There's a race condition where knotd sometimes ends up in a loop when waiting for DS RRs to get propagated to the respective parent zones. I've never seen this before, only on Ubuntu (afaik), however, reloading the zone configuration after committing all changes seems to mitigate this (or at least I can't reproduce the issue anymore). PTAL.

File: test/units/testsuite-75.sh [hunk 1]
```diff
@@ -76,9 +79,7 @@ if knotc zone-get test. onlinesign.test. ds | grep .; then
     # Drop any old DS records, if present (e.g. on test re-run)
     knotc zone-unset test. onlinesign.test. ds
 fi
-knotc zone-commit test.
 # Propagate the new DS records
-knotc zone-begin test.
```
Comment (yuwata) [inline test/units/testsuite-75.sh]: The change in the third commit should be moved to the first commit.

Comment (mrc0mmand) [reply test/units/testsuite-75.sh]: Whoops, indeed, thanks.

**Comment by surajkrishnan14** [comment]
> Interesting. There's a race condition where knotd sometimes ends up in a loop when waiting for DS RRs to get propagated to the respective parent zones. I've never seen this before on Arch/Fedora, only now on Ubuntu (afaik). However, reloading the zone configuration after committing all changes seems to mitigate this (or at least I can't reproduce the issue anymore). PTAL.

Thank you - I tried patch 8b1fd82222b6790ddce9c22b5251b473abe085b7 on this PR, and finally got the test suite passing on my ubuntu 22.04 VM! :-)

**Review by surajkrishnan14 — APPROVED** [review]

**Pull request merged**